types/node-rsa/index.d.ts

```diff
@@ -0,0 +1,197 @@
+// Type definitions for node-rsa 0.4
+// Project: https://github.com/rzcoder/node-rsa
+// Definitions by: Ali Taheri <https://github.com/alitaheri>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+/// <reference types="node" />
+
+declare class NodeRSA {
+    /**
+     * Generate new key with length specified.
+     */
+    constructor(key?: NodeRSA.KeyBits);
```

Comment:
It would depend on what you pass in first.

if it's `{b: 512}` then it will say new key with no extra arguments.

if it's a buffer object or a string or an object with components (`{n: ..., e: ..., d: ..., ...}`) then it will say load and ask for other two optional arguments too.

Format is optional according to documentations. it will assume some defaults.

P.S. Thanks for helping making this better 👍 👍